src/store/utility.ts

```diff
@@ -478,3 +484,46 @@ export const isOnCommandOrCtrlKeyDown = (event: {
   metaKey: boolean;
   ctrlKey: boolean;
 }) => (isMac && event.metaKey) || (!isMac && event.ctrlKey);
+
+/**
+ * スタイルがシングエディタで利用可能なスタイルかどうかを判定します。
+ */
+export const isSingingStyle = (styleInfo: StyleInfo) => {
+  return (
+    styleInfo.styleType === "humming" ||
+    styleInfo.styleType === "sing" ||
+    styleInfo.styleType === "sing_teacher"
+  );
+};
+
+/**
+ * CharacterInfoの配列を、指定されたスタイルタイプでフィルタリングします。
+ */
+export const filterCharacterInfosByStyleType = (
```

Comment:
これは結構危ない挙動をしそうに感じました！
filter: talkしたときにundefinedが入ってくるのが予想外だと感じる人もいそうです。

まあ、ドキュメントを足すのが良いかなと。
```suggestion
/**
 * CharacterInfoの配列を、指定されたスタイルタイプでフィルタリングします。
 * singerLikeはソング系スタイルのみを残します。
 * talkはソング系スタイルをすべて除外します。
 */
export const filterCharacterInfosByStyleType = (
```

あと今はsingerLikeかそれ以外かを取得できる関数があれば十分なので、先に十徳ナイフを作って結局使われないみたいなのが容易に起こる気がします。（YAGNI原則に反する）
とりあえず機能削減するのもありかもです。